lib/smart_answer_flows/simplified-expenses-checker.rb

```diff
@@ -334,6 +334,17 @@ def define
           calculator.simple_home_costs
         end
 
+        precalculate :expenses_or_allowances do
+          calculator.selected_allowance
+        end
+
+        precalculate :vehicle_write_offs do
```

Comment:
I have used precalculate blocks in order not to make more changes.. I think that leaving them alone has made your review a lot more easier